src/lib/pmd/OutputProcessor.ts

```diff
@@ -0,0 +1,110 @@
+import { Logger, Messages } from '@salesforce/core';
+import { AsyncCreatable } from '@salesforce/kit';
+import { uxEvents } from '../ScannerEvents';
+
+
+Messages.importMessagesDirectory(__dirname);
+const messages = Messages.loadMessages('@salesforce/sfdx-scanner', 'EventKeyTemplates');
+
+export type PmdCatalogEvent = {
+    key: string;
+    args: string[];
+    type: string;
+    log: string;
+    handler: string;
+    verbose: boolean;
+    time: number;
+};
+
+/**
+ * Helps with processing output from PmdCatalog java module and converting messages into usable events
+ */
+export class OutputProcessor extends AsyncCreatable {
+
+    private logger!: Logger;
+    private messageLogger!: Logger;
```

Comment:
Is the `!` necessary?

Reply:
I picked this up from Thomas' example of initializing a logger. I'm not sure why, but my guess would be that this would prevent initialization error when typescript is compiled with `--strict` clause. Especially because this wouldn't be initialized in a normal constructor way, but rather through the `init()` method.
https://2ality.com/2020/03/class-definitions-typescript.html#strict-property-initialization